lib/internal/modules/cjs/loader.js

```diff
@@ -412,7 +385,7 @@ function readPackageScope(checkPath) {
     if (StringPrototypeEndsWith(checkPath, sep + 'node_modules'))
       return false;
     const pjson = _readPackage(checkPath + sep);
-    if (pjson) return {
+    if (pjson.exists) return {
```

Comment:
Why have `exists` an additional member, rather than null/undefined?

Reply:
There are some default values that we need to check, and `exists` makes it easier to understand the flow. Performance wise, it negligible. `exists` field was implemented in ESM, but not in CJS. I've used ESM's syntax as the baseline. So, this is not something new.